Samples/iOS-Swift/PerformanceBenchmarks/SentrySDKPerformanceBenchmarkTests.m

```diff
@@ -0,0 +1,134 @@
+#import "SentryProcessInfo.h"
+#import <XCTest/XCTest.h>
+#import <objc/runtime.h>
+
+// To get around the 15 minute timeout per test case on Sauce Labs.
+static NSUInteger SentrySDKPerformanceBenchmarkTestCases = 5;
+static NSUInteger SentrySDKPerformanceBenchmarkIterationsPerTestCase = 4;
+
+// All results are aggregated to analyse after completing the separate,
+// dynamically generated test cases
+static NSMutableArray *allResults;
+static BOOL checkedAssertions = NO;
+
+@interface SentrySDKPerformanceBenchmarkTests : XCTestCase
+
+@end
+
+@implementation SentrySDKPerformanceBenchmarkTests
+
+/**
+ * Dynamically add a test method to an XCTestCase class.
+ * @see https://www.gaige.net/dynamic-xctests.html
+ */
++ (BOOL)addInstanceMethodWithSelectorName:(NSString *)selectorName block:(void (^)(id))block
+{
+    NSParameterAssert(selectorName);
+    NSParameterAssert(block);
+
+    // See
+    // http://stackoverflow.com/questions/6357663/casting-a-block-to-a-void-for-dynamic-class-method-resolution
+    id impBlockForIMP = (__bridge id)(__bridge void *)(block);
+    IMP myIMP = imp_implementationWithBlock(impBlockForIMP);
+    SEL selector = NSSelectorFromString(selectorName);
+    return class_addMethod(self, selector, myIMP, "v@:");
+}
+
++ (void)initialize
+{
+    allResults = [NSMutableArray array];
+    for (NSUInteger i = 0; i < SentrySDKPerformanceBenchmarkTestCases; i++) {
+        [self addInstanceMethodWithSelectorName:[NSString stringWithFormat:@"testCPUBenchmark%lu",
+                                                          (unsigned long)i]
+                                          block:^(XCTestCase *testCase) {
+                                              [allResults
+                                                  addObjectsFromArray:[self _testCPUBenchmark]];
+                                          }];
+    }
```

Comment:
`m`: Why do we need to dynamically add test cases? Why can't we use one single test case and use a for loop for SentrySDKPerformanceBenchmarkTestCases? If there is a good reason, please also add your answer as a comment.

Reply:
There's an unchangeable timeout for test case methods in sauce labs of 15 minutes per test case method. This allows us to fine tune the amount of work that is done in any individual test case method to stay under that limit. I'll add comments/docs about this.